src/Interpreters/executeQuery.cpp

```diff
@@ -233,7 +233,7 @@ inline UInt64 time_in_seconds(std::chrono::time_point<std::chrono::system_clock>
     return std::chrono::duration_cast<std::chrono::seconds>(timepoint.time_since_epoch()).count();
 }
 
-static void onExceptionBeforeStart(const String & query_for_logging, ContextPtr context, UInt64 current_time_us, ASTPtr ast)
+static void onExceptionBeforeStart(const String & query_for_logging, ContextPtr context, UInt64 current_time_us, ASTPtr ast, std::shared_ptr<OpenTelemetrySpanHolder>& query_span)
```

Comment:
Could pass by const-ref. Otherwise the reader wonders why we don't check for nullptr in l. 294-298.